### PR TITLE
feat(registrar): allow bulk invitations

### PR DESCRIPTION
**Pull request by Johaney-s**
* similarily to generating sponsored members, we want to allow inviting users by provided csv

### CODE REVIEW

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 1]
```diff
@@ -137,6 +137,41 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	 * Send invitations with link to VO / Group application form from provided csv data
+	 *
+	 * If VO or Group have non-empty attribute urn:perun:[vo/group]:attribute-def:def:applicationURL
+	 * content is used as link to application form. Otherwise link is automatically generated based on
+	 * required AUTHZ in template and registrar url set in /etc/perun/perun-registrar.properties.
+	 *
+	 * General errors result in exception being thrown, single failures are skipped and added to result
+	 *
+	 * @param invitationData csv file values separated by semicolon ';'. Only [email; name] or [email] is valid format.
+	 * 		example: ["mail@mail.cz", "mail2@mail.cz;user2"]
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:151]: You can use `@exampleParam invitationData ["mail@mail.cz", "mail2@mail.cz;user2"]` to include it in example method call in generated API docs.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 2]
```diff
@@ -137,6 +137,41 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	 * @param vo VO to link form to
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:152]: `vo` param type should be `int` and not `VO` as we read it as integer in the code. Also I would prefer using `voId`, just like in standard `sendInvitation()` methods. Same goes for `group` param.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 3]
```diff
@@ -137,6 +137,41 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	 * @param group Group to link form to
+	 * @param language Language used in notification (if not specified, VO settings is used, if not set, "en" is used).
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:154]: Please include param type `String` for the language.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 4]
```diff
@@ -137,6 +137,41 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	 * Send invitations with link to VO / Group application form from provided csv data
+	 *
+	 * If VO or Group have non-empty attribute urn:perun:[vo/group]:attribute-def:def:applicationURL
+	 * content is used as link to application form. Otherwise link is automatically generated based on
+	 * required AUTHZ in template and registrar url set in /etc/perun/perun-registrar.properties.
+	 *
+	 * General errors result in exception being thrown, single failures are skipped and added to result
+	 *
+	 * @param invitationData csv file values separated by semicolon ';'. Only [email; name] or [email] is valid format.
+	 * @exampleParam ["mail@mail.cz", "mail2@mail.cz;user2"]
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:151]: I think you need to include `invitationData` for the parser to know for which parameter is this example used.

File: perun-openapi/openapi.yml [hunk 5]
```diff
@@ -15052,6 +15052,39 @@ paths:
...
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              title: InputInvitationsFromCsv
+              description: "input for sending invitations from csv"
+              type: object
+              required:
+                - vo
```
Comment (zlamalp) [inline perun-openapi/openapi.yml:15080]: This should be probably `voId` too.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java (hunk 4 above)
Comment (Johaney-s) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java]: I see, added

**Review by cuadradek — APPROVED** [review]

**Pull request merged**